app/src/main/java/com/github/se/polyfit/ui/screen/GraphScreen.kt

```diff
@@ -0,0 +1,126 @@
+package com.github.se.polyfit.ui.screen
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
+import co.yml.charts.ui.linechart.LineChart
+import com.github.se.polyfit.R
+import com.github.se.polyfit.ui.components.DropDownMenu
+import com.github.se.polyfit.ui.components.lineChartData
+import com.github.se.polyfit.ui.components.scaffold.SimpleTopBar
+import com.github.se.polyfit.ui.viewModel.DataToPoints
+import com.github.se.polyfit.ui.viewModel.DateList
+import com.github.se.polyfit.ui.viewModel.GraphViewModel
+import com.github.se.polyfit.ui.viewModel.SortPoints
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun FullGraphScreen(viewModel: GraphViewModel = hiltViewModel<GraphViewModel>()) {
+
+  val isTestEnvironment = System.getProperty("isTestEnvironment") == "true"
+  val isExpanded = remember { mutableStateOf(false) }
```

Comment:
what is the reason behind having this in this file rather than DropDown? It seems like it is only used and mutated there, and could be local to that file?

Reply:
You're right